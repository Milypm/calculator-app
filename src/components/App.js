import React, { useState } from 'react';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';
import '../style/style.css';

const App = () => {
  const [total, setTotal] = useState(null);
  const [next, setNext] = useState('0');
  const [operation, setOperation] = useState(null);

  const handleClick = (btnName) => {
    const obj = { total, next, operation };
    const calcObj = calculate(obj, btnName);
    setTotal(calcObj.total);
    setNext(calcObj.next);
    setOperation(calcObj.operation);
  };

  return (
    <>
      <div>
        <div className="calc-container display-container">
          <p>{next}</p>
        </div>
        <ButtonPanel clickHandler={handleClick} />
      </div>
    </>
  );
};

export default App;
