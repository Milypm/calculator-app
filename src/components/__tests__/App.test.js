import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import App from '../App';

describe('App component', () => {
  test('renders App component', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
