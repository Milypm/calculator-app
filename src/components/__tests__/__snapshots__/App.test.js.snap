// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App component 1`] = `
<div>
  <div
    className="calc-container display-container"
  >
    <p>
      0
    </p>
  </div>
  <div
    className="calc-container btnPanel-container"
  >
    <div
      className="btns-row"
    >
      <button
        className="button-container grey normal"
        onClick={[Function]}
        type="button"
      >
        <span>
          AC
        </span>
      </button>
      <button
        className="button-container grey normal"
        onClick={[Function]}
        type="button"
      >
        <span>
          +/-
        </span>
      </button>
      <button
        className="button-container grey normal"
        onClick={[Function]}
        type="button"
      >
        <span>
          %
        </span>
      </button>
      <button
        className="button-container orange normal"
        onClick={[Function]}
        type="button"
      >
        <span>
          รท
        </span>
      </button>
    </div>
    <div
      className="btns-row"
    >
      <button
        className="button-container grey normal"
        onClick={[Function]}
        type="button"
      >
        <span>
          7
        </span>
      </button>
      <button
        className="button-container grey normal"
        onClick={[Function]}
        type="button"
      >
        <span>
          8
        </span>
      </button>
      <button
        className="button-container grey normal"
        onClick={[Function]}
        type="button"
      >
        <span>
          9
        </span>
      </button>
      <button
        className="button-container orange normal"
        onClick={[Function]}
        type="button"
      >
        <span>
          x
        </span>
      </button>
    </div>
    <div
      className="btns-row"
    >
      <button
        className="button-container grey normal"
        onClick={[Function]}
        type="button"
      >
        <span>
          4
        </span>
      </button>
      <button
        className="button-container grey normal"
        onClick={[Function]}
        type="button"
      >
        <span>
          5
        </span>
      </button>
      <button
        className="button-container grey normal"
        onClick={[Function]}
        type="button"
      >
        <span>
          6
        </span>
      </button>
      <button
        className="button-container orange normal"
        onClick={[Function]}
        type="button"
      >
        <span>
          -
        </span>
      </button>
    </div>
    <div
      className="btns-row"
    >
      <button
        className="button-container grey normal"
        onClick={[Function]}
        type="button"
      >
        <span>
          1
        </span>
      </button>
      <button
        className="button-container grey normal"
        onClick={[Function]}
        type="button"
      >
        <span>
          2
        </span>
      </button>
      <button
        className="button-container grey normal"
        onClick={[Function]}
        type="button"
      >
        <span>
          3
        </span>
      </button>
      <button
        className="button-container orange normal"
        onClick={[Function]}
        type="button"
      >
        <span>
          +
        </span>
      </button>
    </div>
    <div
      className="btns-row"
    >
      <button
        className="button-container grey large"
        onClick={[Function]}
        type="button"
      >
        <span>
          0
        </span>
      </button>
      <button
        className="button-container grey normal"
        onClick={[Function]}
        type="button"
      >
        <span>
          .
        </span>
      </button>
      <button
        className="button-container orange normal"
        onClick={[Function]}
        type="button"
      >
        <span>
          =
        </span>
      </button>
    </div>
  </div>
</div>
`;
